### PR TITLE
184557824 Ft change Sale status 

### PR DESCRIPTION
**Pull request by UwaweTera**
### What does this PR do?

 change a product-order status

### Description of Task to be completed?

 change a product-order status . accepted, declined, delivered : default: pending-review

### How should this be manually tested?

Provides steps on how this feature can tested.

npm install
npm start

### Any background context you want to provide?

none

### What are the relevant pivotal tracker stories?

https://www.pivotaltracker.com/n/projects/2628461/stories/184557824

### Screenshots (if appropriate)
none

### Questions:

# Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code is formatted correctly
- [x] My changes doesn't have any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Ntarecedrick — APPROVED** [review]

**Review by teerenzo — APPROVED** [review]

File: src/services/paymentService.js [hunk 1]
```diff
@@ -38,24 +36,22 @@ class PaymentService {
...
         cancel_url: `${appUrl}/api/v1/payment/cancel-callback?token=${token}`,
       });
-      res
-        .status(200)
-        .json({token, paymentId: session.id, url: session.url });
+      res.status(200).json({ token, paymentId: session.id, url: session.url });
     } catch (error) {
-        /* istanbul ignore next */
+      /* istanbul ignore next */
       res.status(500).json({ error });
```
Comment (jimmymut) [inline src/services/paymentService.js:42]: here, I think error.message would be much better

File: src/services/paymentService.js [hunk 2]
```diff
@@ -79,8 +75,8 @@ class PaymentService {
       }
       return session.payment_status;
     } catch (error) {
-        /* istanbul ignore next */
-        res.status(500).json({ error });
+      /* istanbul ignore next */
+      res.status(500).json({ error });
```
Comment (jimmymut) [inline src/services/paymentService.js:79]: same here. error: error.message

File: src/services/paymentService.js (hunk 1 above)
Comment (UwaweTera) [reply src/services/paymentService.js:42]: @jimmymut  I think this code belongs to @teerenzo  worked on payment

File: src/services/paymentService.js (hunk 2 above)
Comment (UwaweTera) [reply src/services/paymentService.js:79]: Also this code belongs to @teerenzo may be I will try to reach on him

File: src/services/paymentService.js (hunk 1 above)
Comment (jimmymut) [reply src/services/paymentService.js:42]: I realized that after I added the comment

File: src/services/paymentService.js (hunk 2 above)
Comment (jimmymut) [reply src/services/paymentService.js:79]: yaep

File: src/controllers/saleController.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable prettier/prettier, prefer-destructuring, require-jsdoc */
+import SaleServices from "../services/saleService";
+
+class SaleController {
+    static async updateSaleStatus(req, res){
+        try {     
+            const UpdatedSale = await SaleServices.updateSaleStatus(req);
+            return res.status(200).json({
+                status: 200,
+                Sale: UpdatedSale
```
Comment (Eli250) [inline src/controllers/saleController.js:10]: Return the sale status message instead of the entire sale model.

**Review by Eli250 — APPROVED** [review]

**Pull request merged**